awswrangler/_config.py

```diff
@@ -37,22 +39,27 @@ class _ConfigArg(NamedTuple):
     "workgroup": _ConfigArg(dtype=str, nullable=False, enforced=True),
     "chunksize": _ConfigArg(dtype=int, nullable=False, enforced=True),
     # Endpoints URLs
-    "s3_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True),
-    "athena_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True),
-    "sts_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True),
-    "glue_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True),
-    "redshift_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True),
-    "kms_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True),
-    "emr_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True),
-    "lakeformation_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True),
-    "dynamodb_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True),
-    "secretsmanager_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True),
-    "timestream_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True),
+    "s3_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True, loaded=True),
+    "athena_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True, loaded=True),
+    "sts_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True, loaded=True),
+    "glue_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True, loaded=True),
+    "redshift_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True, loaded=True),
+    "kms_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True, loaded=True),
+    "emr_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True, loaded=True),
+    "lakeformation_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True, loaded=True),
+    "dynamodb_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True, loaded=True),
+    "secretsmanager_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True, loaded=True),
+    "timestream_endpoint_url": _ConfigArg(dtype=str, nullable=True, enforced=True, loaded=True),
     # Botocore config
     "botocore_config": _ConfigArg(dtype=botocore.config.Config, nullable=True),
-    "verify": _ConfigArg(dtype=str, nullable=True),
+    "verify": _ConfigArg(dtype=str, nullable=True, loaded=True),
     # Distributed
-    "distributed": _ConfigArg(dtype=bool, nullable=True),
+    "execution_engine": _ConfigArg(
+        dtype=str, nullable=True, loaded=True, default="ray" if importlib.util.find_spec("ray") else "python"
+    ),
+    "memory_format": _ConfigArg(
+        dtype=str, nullable=True, loaded=True, default="modin" if importlib.util.find_spec("modin") else "pandas"
+    ),
     "address": _ConfigArg(dtype=str, nullable=True),
```

Comment:
I did think about it but found it pretty ugly (i.e. redundant) to be honest and of little value. We shouldn't be in a situation where more than one engine is used at once so the env variables wouldn't clash

Reply:
That being said, it might confusing for users to know which variable to use for which engine without a description somewhere, so perhaps there is value... Don't know, would want to hear from others too